PhysicsTools/PatAlgos/python/slimming/extraJets_MuEGFixMoriond2017.py

```diff
@@ -0,0 +1,119 @@
+                           triggeringCollections=badMuons,
+                           )
+            )
+
+def reduceFinalJetCollection(process, jetCollection, badMuons):
+    label = jetCollection.label()
+    setattr(process, label+"AllEvents", jetCollection.clone())
+    process.globalReplace(label,
+            cms.EDProducer("PATJetCollectionReducer",
+                           writeEmptyCollection = cms.bool(False),
+                           jetCollection = cms.InputTag(label+"AllEvents"),
+                           triggeringCollections=badMuons,
+                           ) 
+            )
+    
+
+def backupJetsFirstStep(process):
+    """Take snapshots of the sequences before we change the PFCandidates"""
+    process.originalAK4JetSequence = listDependencyChain(process, process.slimmedJets, ('particleFlow', 'muons'))
+    backupAK4JetSequence = cloneProcessingSnippet(process, process.originalAK4JetSequence, "Backup")
+    process.originalAK4PuppiJetSequence = listDependencyChain(process, process.slimmedJetsPuppi, ('particleFlow', 'muons'))
+    backupAK4PuppiJetSequence = cloneProcessingSnippet(process, process.originalAK4PuppiJetSequence, "Backup")
+    process.originalAK8JetSequence = listDependencyChain(process, process.slimmedJetsAK8, ('particleFlow', 'muons'))
+    backupAK8JetSequence = cloneProcessingSnippet(process, process.originalAK8JetSequence, "Backup")
+    return { 'AK4':backupAK4JetSequence, 'AK4Puppi':backupAK4PuppiJetSequence, 'AK8':backupAK8JetSequence }
+
+    
+def backupJetsSecondStep(process, sequences, badMuons, verbose=False):
+    """Deploy the snapshots after the change of PFCandidates"""
+    # put back the old input tags
+    for sequence in sequences.itervalues():
+        massSearchReplaceAnyInputTag(sequence, "pfCandidatesBadMuonsCleaned", "particleFlow")
+        massSearchReplaceAnyInputTag(sequence, "muonsCleaned", "muons")
+    # gate the input collections to avoid re-running most of PAT on good events
+    reduceInputJetCollection(process, process.ak4PFJetsCHSBackup, badMuons)
+    reduceInputJetCollection(process, process.ak4PFJetsPuppiBackup, badMuons)
+    # fix names in the valuemaps
+    process.patJetsBackup.userData.userInts.labelPostfixesToStrip = cms.vstring("Backup",)
+    process.patJetsBackup.userData.userFloats.labelPostfixesToStrip = cms.vstring("Backup",)
+    process.patJetsAK8Backup.userData.userFloats.labelPostfixesToStrip = cms.vstring("Backup",)
+    process.patJetsAK8PuppiBackup.userData.userFloats.labelPostfixesToStrip = cms.vstring("Backup",)    
+    #
+    # now deal with daughter links
+    # for these we can keep the daughters
+    process.slimmedJetsBackup.mixedDaughters = True
+    process.slimmedJetsBackup.packedPFCandidates = cms.InputTag("oldPFCandToPackedOrDiscarded")
+    process.slimmedJetsAK8PFCHSSoftDropSubjetsBackup.mixedDaughters = True
+    process.slimmedJetsAK8PFCHSSoftDropSubjetsBackup.packedPFCandidates = cms.InputTag("oldPFCandToPackedOrDiscarded")
```

Comment:
do these still exist in 90X?
These may need to be removed or put behind an conditional statement